shell/platform/linux/fl_text_input_plugin.cc

```diff
@@ -642,7 +644,8 @@ static gboolean fl_text_input_plugin_filter_keypress_default(
       case GDK_KEY_Return:
       case GDK_KEY_KP_Enter:
       case GDK_KEY_ISO_Enter:
-        if (priv->input_type == kFlTextInputTypeMultiline) {
+        if (priv->input_type == kFlTextInputTypeMultiline &&
+            strcmp(priv->input_action, kFlTextInputActionSend) != 0) {
```

Comment:
should we rather compare to "TextInputAction.newline" because it's probably the only text input action that should insert "\n"?

Reply:
That seems to make sense to me after looking through all of the other [TextInputAction](https://master-api.flutter.dev/flutter/flutter_driver/TextInputAction.html) values.  That's also the default for multiline fields.

Reply:
> should we rather compare to "TextInputAction.newline" because it's probably the only text input action that should insert "\n"?

I wanted to be as conservative as possible because I'm not sure about the meaning of all the TextInputAction values on desktop.
But I agree that it would make sense to insert "\n" only when action is `TextInputAction.newline`.

@justinmc Can you confirm it would be preferable?

Reply:
Yeah I agree 👍.  So to be specific, I think you should keep the multiline check, and instead of checking that it's not kFlTextInputActionSend, check that it *is* the newline action.